python-backend/app/api/mines/incidents/resources/mine_incidents.py

```diff
@@ -295,8 +305,33 @@ def put(self, mine_guid, mine_incident_guid):
                     db.session.delete(doc)
                     db.session.commit()
 
-        try:
-            incident.save()
-        except Exception as e:
-            raise InternalServerError(f'Error when saving: {e}')
+        incident.save()
         return incident
+
+    @classmethod
+    def _handle_recommendations(cls, incident, recommendations):
+        for payload in recommendations:
```

Comment:
Minor naming quibble: the use of the word payload doesn't seem right. It's not as readable as 'recommendation' and not I haven't really seen it used as a generic iterator name vs 'i' or 'item'.

Reply:
Good point. I should have actually replaced `payload` before opening the PR. It was functioning as a placeholder when I first added it and then I became blind to it. I'll update